bc_obps/registration/tests/endpoints/test_useroperator_endpoints.py

```diff
@@ -687,7 +688,77 @@ def test_create_operator_and_user_operator_with_parent_operators(self):
         assert parent_operators[0].operator_index == 1
         assert parent_operators[1].operator_index == 2
 
+    def test_edit_and_archive_parent_operators(self):
+
+        child_operator = operator_baker()
+        user_operator = baker.make(UserOperator, operator=child_operator, user=self.user)
+
+        parent_operator_1 = parent_operator_baker()
+        parent_operator_1.child_operator_id = child_operator.id
+        parent_operator_1.operator_index = 1
+        parent_operator_1.save(update_fields=["child_operator_id", "operator_index"])
+
+        parent_operator_2 = parent_operator_baker()
+        parent_operator_2.child_operator_id = child_operator.id
+        parent_operator_2.operator_index = 2
+        parent_operator_2.save(update_fields=["child_operator_id", "operator_index"])
+
+        mock_payload = {
+            "legal_name": "New Operator",
+            "trade_name": "New Operator Trade Name",
+            "cra_business_number": 963852741,
+            "bc_corporate_registry_number": "adh1234321",
+            "business_structure": BusinessStructure.objects.first().pk,
+            "physical_street_address": "test physical street address",
+            "physical_municipality": "test physical municipality",
+            "physical_province": "BC",
+            "physical_postal_code": "H0H0H0",
+            "mailing_address_same_as_physical": True,
+            "operator_has_parent_operators": True,
+            "parent_operators_array": [
+                {
+                    "po_legal_name": "test po legal name-EDITED",
+                    "po_cra_business_number": 123456789,
+                    "po_bc_corporate_registry_number": "poo7654321",
+                    "po_business_structure": BusinessStructure.objects.first().pk,
+                    "po_website": "https://testpo.com",
+                    "po_physical_street_address": "test po physical street address",
+                    "po_physical_municipality": "test po physical municipality",
+                    "po_physical_province": "ON",
+                    "po_physical_postal_code": "H0H0H0",
+                    "po_mailing_address_same_as_physical": True,
+                    "operator_index": 2,
+                },
+            ],
+        }
+        response = TestUtils.mock_put_with_auth_role(
+            self,
+            'industry_user',
+            content_type_json,
+            mock_payload,
+            f"{self.user_operator_endpoint}/operator/{user_operator.id}",
+        )
+        assert response.status_code == 200
+
+        parent_operators: List[ParentOperator] = child_operator.parent_operators.all()
+
+        assert len(parent_operators) == 1  # archived records are not pulled
```

Comment:
@Sepehr-Sobhani , is it expected that archived records don't show up when doing a selection? I didn't add any code to do this; I just used the existing `set_archive` function

Reply:
Yes, we're using the `TimeStampedModelManager` and that custom manager automatically filters out archived records for every data model inheriting `TimeStampedModel` and we only see the active ones. If we need to access the archived records for any particular data model, we'll need to override this custom manager specifically for that purpose.